Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/InputActionReferencePropertyDrawer.cs

```diff
@@ -0,0 +1,47 @@
+// Note: If not UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS we do not use a custom property drawer and
+//       picker for InputActionReferences but rather rely on default (classic) object picker.
+#if UNITY_EDITOR && UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
+
+using UnityEditor;
+using UnityEditor.Search;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    /// <summary>
+    /// Custom property drawer in order to use the "Advanced Picker" from UnityEditor.Search.
+    /// </summary>
+    [CustomPropertyDrawer(typeof(InputActionReference))]
+    internal sealed class InputActionReferencePropertyDrawer : PropertyDrawer
+    {
+        private readonly SearchContext m_Context = UnityEditor.Search.SearchService.CreateContext(new[]
+        {
+            InputActionReferenceSearchProviders.CreateInputActionReferenceSearchProviderForAssets(),
+            InputActionReferenceSearchProviders.CreateInputActionReferenceSearchProviderForProjectWideActions(),
+        }, string.Empty, SearchConstants.PickerSearchFlags);
+
+
+        public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+        {
+            // Sets the property to null if the action is not found in the asset.
+            ValidatePropertyWithDanglingInputActionReferences(property);
```

Comment:
Why do we need to do this? Note that this also introduces a different behavior than for normal .inputaction assets which resolve to "Missing (Input Action Reference)" when the referenced ScriptableObject is Destroyed and marked available for GC. I assume this will resolve to null if its a dangling instance from broken Undo and maybe its better than having a "dangling reference" but wouldn't this also prevent recovery via Undo?

Reply:
It seems like this will trigger a full iteration of all action maps and all actions on every redraw right?

Reply:
If this is not here and an Input Action is removed from the asset, it continues to reference a removed input action.
But this code has the implications you just mentioned. Ideally, we would like to remove this so that this property is updated once the action is removed from the asset. 

This is only needed for references of the project-wide actions asset. Normal assets work just fine.

Reply:
Ok, I just checked this in the latest 2022.3.11f, and this code seems to not be needed anymore. I can remove this code as it will have the same behavior. This is likely a leftover from doing most of the work with 2022.3.0f.....

Reply:
Removed

Reply:
Ok, I was testing again, and without this code a removed action was still being referenced in the property. And we want to show None/Missing. I reintroduced this but now only checking for Project Wide Actions. I captured a task to remove this: https://jira.unity3d.com/browse/ISX-1683